reactor-netty-http/src/test/java/reactor/netty/http/client/WebsocketClientOperationsTest.java

```diff
@@ -54,73 +58,77 @@ private void failOnClientServerError(
-				                        if (token >= 400) {
-				                            return res.status(token).send();
-				                        }
-				                        return res.sendWebsocket((i, o) -> o.sendString(Mono.just("test")),
-				                                WebsocketServerSpec.builder().protocols(serverSubprotocol).build());
-				                    }))
-				          .wiretap(true)
-				          .bindNow();
+						.port(0)
+						.route(routes ->
+								routes.post("/login", (req, res) -> res.status(serverStatus).sendHeaders())
+										.get("/ws", (req, res) -> {
+											int token = Integer.parseInt(req.requestHeaders().get("Authorization"));
+											if (token >= 400) {
+												return res.status(token).send();
+											}
+											return res.sendWebsocket((i, o) -> o.sendString(Mono.just("test")),
+													WebsocketServerSpec.builder().protocols(serverSubprotocol).build());
+										}))
+						.wiretap(true)
+						.bindNow();
 
 		Flux<String> response =
 				HttpClient.create()
-				          .port(httpServer.port())
-				          .wiretap(true)
-				          .headersWhen(h -> login(httpServer.port()).map(token -> h.set("Authorization", token)))
-				          .websocket(WebsocketClientSpec.builder().protocols(clientSubprotocol).build())
-				          .uri("/ws")
-				          .handle((i, o) -> i.receive().asString())
-				          .log()
-				          .switchIfEmpty(Mono.error(new Exception()));
+						.port(httpServer.port())
+						.wiretap(true)
+						.headersWhen(h -> login(httpServer.port()).map(token -> h.set("Authorization", token)))
+						.websocket(WebsocketClientSpec.builder().protocols(clientSubprotocol).build())
+						.uri("/ws")
+						.handle((i, o) -> i.receive().asString())
+						.log()
+						.switchIfEmpty(Mono.error(new Exception()));
 
 		StepVerifier.create(response)
-		            .expectError(WebSocketHandshakeException.class)
-		            .verify();
+				.expectError(WebSocketHandshakeException.class)
+				.verify();
 
 		httpServer.disposeNow();
 	}
 
 	private Mono<String> login(int port) {
 		return HttpClient.create()
-		                 .port(port)
-		                 .wiretap(true)
-		                 .post()
-		                 .uri("/login")
-		                 .responseSingle((res, buf) -> Mono.just(res.status().code() + ""));
+				.port(port)
+				.wiretap(true)
+				.post()
+				.uri("/login")
+				.responseSingle((res, buf) -> Mono.just(res.status().code() + ""));
 	}
 
 	@Test
-	public void testConfigureWebSocketVersion(){
+	public void testConfigureWebSocketVersion() {
+		ArgumentCaptor<HttpHeaders> argumentCaptor = ArgumentCaptor.forClass(HttpHeaders.class);
+
 		DisposableServer httpServer = HttpServer.create()
 				.port(0)
-				.handle((in, out) -> out.sendWebsocket((i, o) -> o.sendString(Mono.just("test"))))
+				.handle((in, out) -> {
+					assertThat(in.requestHeaders().get("sec-websocket-version")).isEqualTo("8");
```

Comment:
does the `AssertionError` escape the `handle` and actually fails the test?
I had a doubt and verified locally, it seems to log the assertion error but then hangs...

another way would be for the `sendString` below to return the content of the header, and then we can assert it in the `HttpClient...block()` call, as the body of the response.

Reply:
just to be clear: the test passes in its current form _because the assertion succeeds_. If you change it to `isEqualTo("foo")` an `AssertionError` is logged but the test then hangs.

Reply:
also, it seems you don't need Mockito at all anymore